WordPress/gutenberg
# Migrate classic menus to block-based menus on theme switch

## Description

Adds an action that acts on `theme_switch` and:
* Short-circuits if the new theme isn't a FSE theme
* Finds all the menus assigned to a location in the old theme
* Migrates each to a wp_navigation post supported by the navigation block
* Updates `fse_navigation_areas` to reflect these migrated menus

`gutenberg_migrate_nav_on_theme_switch` is the only new function, the other ones were copied over from the navigation block. It would be great to remove the duplication in a follow-up PR.

## How has this been tested?

* Switch to any classic theme
* Setup a few menus in a few locations, the locations should be called either primary, secondary, or tertiary
* Switch to tt1-blocks
* Confirm the menus were migrated as expected
